dao/src/main/java/se/fortnox/reactivewizard/db/DbProxy.java

```diff
@@ -113,19 +117,37 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
                 // Need to get the actual interface method in order to get updated annotations
                 method = ReflectionUtil.getRedefinedMethod(method);
             }
+
             DbStatementFactory statementFactory = dbStatementFactoryFactory.createStatementFactory(method);
             PagingOutput pagingOutput = new PagingOutput(method);
             reactiveStatementFactory = new ReactiveStatementFactory(
                     statementFactory,
                     pagingOutput,
                     createMetrics(method),
                     databaseConfig,
-                    converterFromFlux(method.getReturnType()));
+                    converterFromFlux(method),
+                    method);
             statementFactories.put(method, reactiveStatementFactory);
         }
+
         return reactiveStatementFactory.create(args, connectionScheduler);
     }
 
+    private static Function<Flux, Object> converterFromFlux(Method method) {
+        Class<?> returnType = method.getReturnType();
+
+        if (Flux.class.isAssignableFrom(returnType)) {
+            return flux -> flux;
+        } else if (Mono.class.isAssignableFrom(returnType)) {
+            return Mono::from;
```

Comment:
Replace with `Flux::.singleOrEmpty` and drop class `DaoFluxSink`?

Reply:
We lose other important behavior, such as not sending the error if element has been emitted. I've tried to find a one-liner solution but I could not find such solution. Using MonoSink would work but we would be moving code from the custom sink into the MonoSink.